TombEngine/Objects/Generic/Switches/airlock_switch.cpp

```diff
@@ -0,0 +1,82 @@
+#include "framework.h"
+#include "Game/control/control.h"
+#include "Specific/Input/Input.h"
+#include "Specific/level.h"
+#include "Game/Lara/lara.h"
+#include "Game/Lara/lara_helpers.h"
+#include "Objects/Generic/Switches/generic_switch.h"
+#include "Game/animation.h"
+#include "Game/collision/collide_item.h"
+#include "Game/items.h"
+
+using namespace TEN::Input;
+
+namespace TEN::Entities::Switches
+{
+	ObjectCollisionBounds LockSwitchBounds =
+	{
+		GameBoundingBox::Zero,
+		std::pair(
+			EulerAngles(ANGLE(-10.0f), ANGLE(-30.0f), ANGLE(-10.0f)),
+			EulerAngles(ANGLE(10.0f), ANGLE(30.0f), ANGLE(10.0f))
+		)
+	};
+
+	auto AirLockSwitchPos = Vector3i::Zero;
+
+	void AirLockSwitchCollision(short itemNumber, ItemInfo* laraItem, CollisionInfo* coll)
+	{
+		auto* laraInfo = GetLaraInfo(laraItem);
```

Comment:
New file, chance to use references from the start.

Reply:
I didn't understood this one.

Are you suggesting to use references instead of pointers?

```
void AirLockSwitchCollision(short itemNumber, ItemInfo& laraItem, CollisionInfo* coll)
{
	auto laraInfo = GetLaraInfo(laraItem);
```

Reply:
Yes, I mean that with new stuff we can readily adopt references. So these would be:
`auto& laraInfo = *GetLaraInfo(laraItem);`
`auto& switchItem = g_Level.Items[itemNumber];`